core/src/main/java/org/jdbi/v3/core/argument/EnumArgumentFactory.java

```diff
@@ -49,7 +51,18 @@
     }
 
     private static <E extends Enum<E>> Optional<Argument> byName(E value, ConfigRegistry config) {
-        return makeArgument(Types.VARCHAR, String.class, value, E::name, config);
+        return makeArgument(Types.VARCHAR, String.class, value, e -> {
+            final Field field;
+            try {
+                field = e.getDeclaringClass().getField(e.name());
+            } catch (final NoSuchFieldException ex) {
+                throw new RuntimeException(ex);
```

Comment:
please use our `Unchecked.function` for this sort of thing (inspired by jool, if you're familiar with that). I assume both methods `throw`, so you'll have to `uncheck` a lambda rather than the preferred method ref.

Reply:
That’s what I did first, but then the extra `.get()` on the Supplier annoyed me. Is it okay to extend `Unchecked` so it directly returns the `CheckedCallable`’s result?

Reply:
superfluous-looking `get` calls on `Unchecked` usually means you should actually be using `function` rather than `supplier`